scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/quartz/AppScalingScheduleJobTest.java

```diff
@@ -0,0 +1,492 @@
+import org.cloudfoundry.autoscaler.scheduler.dao.ActiveScheduleDao;
+import org.cloudfoundry.autoscaler.scheduler.entity.ActiveScheduleEntity;
+import org.cloudfoundry.autoscaler.scheduler.util.JobActionEnum;
+import org.cloudfoundry.autoscaler.scheduler.util.ScalingEngineUtil;
+import org.cloudfoundry.autoscaler.scheduler.util.ScheduleJobHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.ScheduleTypeEnum;
+import org.cloudfoundry.autoscaler.scheduler.util.TestDataCleanupHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.TestDataSetupHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.error.DatabaseValidationException;
+import org.cloudfoundry.autoscaler.scheduler.util.error.MessageBundleResourceHelper;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.quartz.JobBuilder;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerKey;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@ActiveProfiles(value = { "ActiveScheduleDaoMock", "ScalingEngineUtilMock" })
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class AppScalingScheduleJobTest {
+
+	@Mock
+	private Appender mockAppender;
+
+	@Captor
+	private ArgumentCaptor<LogEvent> logCaptor;
+
+	@Autowired
+	private MessageBundleResourceHelper messageBundleResourceHelper;
+
+	@Autowired
+	private Scheduler scheduler;
+
+	@Autowired
+	private ActiveScheduleDao activeScheduleDao;
+
+	@Autowired
+	private TestDataCleanupHelper testDataCleanupHelper;
+
+	@Value("${autoscaler.scalingengine.url}")
+	private String scalingEngineUrl;
+
+	@Autowired
+	private ScalingEngineUtil scalingEngineUtil;
+
+	@Value("${scalingenginejob.refire.interval}")
+	private static int originalJobRefireInterval;
+
+	@Value("${scalingenginejob.refire.maxcount}")
+	private static long originalJobRefireMaxCount;
+
+	@Before
+	public void before() throws SchedulerException {
+
+		testDataCleanupHelper.cleanupData(scheduler);
+
+		Mockito.reset(activeScheduleDao);
+		Mockito.reset(scalingEngineUtil);
+		Mockito.reset(mockAppender);
+
+		Mockito.when(mockAppender.getName()).thenReturn("MockAppender");
+		Mockito.when(mockAppender.isStarted()).thenReturn(true);
+		Mockito.when(mockAppender.isStopped()).thenReturn(false);
+
+		setLogLevel(Level.INFO);
+	}
+
+	@AfterClass
+	public static void afterClass() {
+		System.setProperty("scalingenginejob.refire.interval", String.valueOf(originalJobRefireInterval));
```

Comment:
Why are we bothering with system properties? Just set the values you want in the before.